systemtest/src/main/java/io/strimzi/systemtest/templates/crd/KafkaMirrorMaker2Templates.java

```diff
@@ -5,48 +5,100 @@
-        ConfigMap metricsCm = TestUtils.configMapFromYaml(TestConstants.PATH_TO_KAFKA_MIRROR_MAKER_2_METRICS_CONFIG, "mirror-maker-2-metrics");
-        kubeClient().createConfigMapInNamespace(namespaceName, metricsCm);
-        return defaultKafkaMirrorMaker2(kafkaMirrorMaker2, name, targetClusterName, sourceClusterName, kafkaMirrorMaker2Replicas, false, sourceNs, targetNs);
+    public static KafkaMirrorMaker2Builder kafkaMirrorMaker2WithMetrics(
+        String namespaceName,
+        String mm2name,
+        String sourceClusterName,
+        String targetClusterName,
+        int kafkaMirrorMaker2Replicas,
+        String sourceNs,
+        String targetNs
+    ) {
+        return defaultKafkaMirrorMaker2(namespaceName, mm2name, sourceClusterName, targetClusterName, kafkaMirrorMaker2Replicas, false, sourceNs, targetNs)
+            .editOrNewSpec()
+                .withNewJmxPrometheusExporterMetricsConfig()
+                    .withNewValueFrom()
+                        .withNewConfigMapKeyRef(CONFIG_MAP_KEY, getConfigMapName(mm2name), false)
+                    .endValueFrom()
+                .endJmxPrometheusExporterMetricsConfig()
+                .editFirstMirror()
+                    .withNewHeartbeatConnector()
+                        .addToConfig("checkpoints.topic.replication.factor", -1)
+                    .endHeartbeatConnector()
+                .endMirror()
+            .endSpec();
     }
 
-    private static KafkaMirrorMaker2Builder defaultKafkaMirrorMaker2(KafkaMirrorMaker2 kafkaMirrorMaker2, String name, String kafkaTargetClusterName, String kafkaSourceClusterName, int kafkaMirrorMaker2Replicas, boolean tlsListener) {
-        return defaultKafkaMirrorMaker2(kafkaMirrorMaker2, name, kafkaTargetClusterName, kafkaSourceClusterName, kafkaMirrorMaker2Replicas, tlsListener, null, null);
+    public static ConfigMap mirrorMaker2MetricsConfigMap(String namespaceName, String mm2Name) {
+        return new ConfigMapBuilder(TestUtils.configMapFromYaml(TestConstants.PATH_TO_KAFKA_MIRROR_MAKER_2_METRICS_CONFIG, "mirror-maker-2-metrics"))
+            .editOrNewMetadata()
+                .withNamespace(namespaceName)
+                .withName(getConfigMapName(mm2Name))
+            .endMetadata()
+            .build();
     }
 
-    private static KafkaMirrorMaker2Builder defaultKafkaMirrorMaker2(KafkaMirrorMaker2 kafkaMirrorMaker2, String name, String kafkaTargetClusterName, String kafkaSourceClusterName, int kafkaMirrorMaker2Replicas, boolean tlsListener, String sourceNs, String targetNs) {
+    private static String getConfigMapName(String mm2Name) {
+        return mm2Name + METRICS_MM2_CONFIG_MAP_SUFFIX;
+    }
+
+    private static KafkaMirrorMaker2Builder defaultKafkaMirrorMaker2(
+        String namespaceName,
+        String mm2Name,
+        String kafkaSourceClusterName,
+        String kafkaTargetClusterName,
+        int kafkaMirrorMaker2Replicas,
+        boolean tlsListener
```

Comment:
I think it's easier to have it like this, because when you look into `defaultKafkaMirrorMaker2`, there is bootstrap server set based on the different namespace etc. So IMHO this is easier, as we can modify it and use all of the other parameters as well (otherwise we would have to determine the bootstrap server in tests, which is maybe not the best approach). Plus we would have to add the trusted certificate there based on some other check, so I guess this is sufficient.

Reply:
ok sounds reasonable